### PR TITLE
In order to delete a user, you must first remove them from any schedu…

### PR DESCRIPTION
**Pull request by JohnTheodore**
…les they're in. You can find the schedules a particular user is in with api/v1/users/user_id/schedules. This is faster than grabbing all schedules and digging through to find a specific user.

### CODE REVIEW

File: pygerduty/__init__.py [hunk 1]
```diff
@@ -505,9 +505,21 @@ def __init__(self, *args, **kwargs):
         Container.__init__(self, *args, **kwargs)
         self.notification_rules = NotificationRules(self.pagerduty, self)
         self.contact_methods = ContactMethods(self.pagerduty, self)
+        self.schedules = UserSchedules(self.pagerduty, self)
+        self.escalation_policies = UserEscalationPolicies(self.pagerduty, self)
         self.log_entries = LogEntries(self.pagerduty, self)
 
 
+class UserSchedules(Collection):
```
Comment (gmjosack) [inline pygerduty/__init__.py]: Did you find that this and `UserEscalationPolicies` was necessary? A quick test with the existing `Schedules` and `EscalationPolicies` collections seemed to work fine.

**Comment by gmjosack** [comment]
Thanks for added this! I've wanted something like this but I haven't seen anything in the API docs. Do you have a reference where you found this was available?

**Comment by JohnTheodore** [comment]
I poked the support people today asking why this isn't in the api. They told me it was, but undocumented. It will be documented 'soon'. :p

File: pygerduty/__init__.py (hunk 1 above)
Comment (JohnTheodore) [inline pygerduty/__init__.py]: So perhaps I've overlooked something, but it didn't seem like the Schedules allowed me to filter for only schedules that include a specific user?

Comment (gmjosack) [inline pygerduty/__init__.py]: To be more clear, adding this

```python
        self.schedules = Schedules(self.pagerduty, self)
        self.escalation_policies = EscalationPolicies(self.pagerduty, self)
```

to the `User` container seemed to work without needing the two new collection classes you added.

**Comment by JohnTheodore** [comment]
ok, I made the changes you suggested and rebased to one commit. Any other changes?

**Comment by gmjosack** [comment]
Looks good. Thanks!

I've pushed this to PyPI as pygerduty-0.32.1

**Pull request merged**